### PR TITLE
Add suspend/resume capabilities to background sending of emails

### PR DESCRIPTION
**Pull request by SeiryuZ**

### CODE REVIEW

File: post_office/models.py [hunk 1]
```diff
@@ -80,6 +83,28 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return u'%s' % self.to
 
+    @classmethod
+    def suspend(self):
+        """ Create a lockfile that suspend background sending of emails.
```
Comment (selwin) [inline post_office/models.py]:
```
Creates a lockfile that suspends background sending of emails.
```

File: post_office/models.py [hunk 2]
```diff
@@ -80,6 +83,28 @@ def __init__(self, *args, **kwargs):
...
+            User is responsible for clearing up the suspend lockfile.
+        """
+        open(SUSPEND_FILEPATH, 'a').close()
+
+    @classmethod
+    def is_suspended(self):
+        """ Check if the suspend lockfile exist or not
```
Comment (selwin) [inline post_office/models.py]: `Check whether email delivery is currently suspended`

File: post_office/management/commands/suspend_mail.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from django.core.management.base import BaseCommand
+
+from ...models import Email
+
+
+class Command(BaseCommand):
```
Comment (selwin) [inline post_office/management/commands/suspend_mail.py]: `suspend_email_delivery`

File: post_office/management/commands/resume_mail.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from django.core.management.base import BaseCommand
+
+from ...models import Email
+
+
+class Command(BaseCommand):
+    help = 'Resume queued email sending.'
+
+    def handle(self, **options):
```
Comment (selwin) [inline post_office/management/commands/resume_mail.py]: `resume_email_delivery`

**Comment by selwin** [comment]
Instead of suspending/resuming email delivery, I have another project which does something like this: ```python
# This means mail will only be delivered during 8am to 8pm
mail.send(foo, bar, delivery_window(time(08, 00), time(20, 00))
```

Let's do this instead.

**Comment by jrief** [comment]
@selwin a good library to handle datetime ranges is here: https://pypi.org/project/DateTimeRange/

**Pull request closed without merging**